{"priority": "major", "kind": "bug", "repository": {"links": {"self": {"href": "data/repositories/philipstarkey/qtutils.json"}, "html": {"href": "#!/philipstarkey/qtutils"}, "avatar": {"href": "data/bytebucket.org/ravatar/{eb85b0e6-11dc-449a-96ec-21935a03d721}ts=python"}}, "type": "repository", "name": "qtutils", "full_name": "philipstarkey/qtutils", "uuid": "{eb85b0e6-11dc-449a-96ec-21935a03d721}"}, "links": {"attachments": {"href": "data/repositories/philipstarkey/qtutils/issues/1/attachments_page=1.json"}, "self": {"href": "data/repositories/philipstarkey/qtutils/issues/1.json"}, "watch": {"href": "https://api.bitbucket.org/2.0/repositories/philipstarkey/qtutils/issues/1/watch"}, "comments": {"href": "data/repositories/philipstarkey/qtutils/issues/1/comments_page=1.json"}, "html": {"href": "#!/philipstarkey/qtutils/issues/1/invoke_in_main-leaks-memory-when-using"}, "vote": {"href": "https://api.bitbucket.org/2.0/repositories/philipstarkey/qtutils/issues/1/vote"}}, "reporter": {"display_name": "Philip Starkey", "uuid": "{48af65db-e5fc-459c-a7eb-52eb1f9a5690}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7B48af65db-e5fc-459c-a7eb-52eb1f9a5690%7D"}, "html": {"href": "https://bitbucket.org/%7B48af65db-e5fc-459c-a7eb-52eb1f9a5690%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/dc318537facc47ebe1ae98a7aabacecfd=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsPS-0.png"}}, "nickname": "pstarkey", "type": "user", "account_id": "557058:52a111e4-40da-4441-9143-417f95f2db97"}, "title": "invoke_in_main leaks memory when using PySide", "component": null, "votes": 0, "watches": 1, "content": {"raw": "Calls to functions in invoke_in_main cause the Python program to leak memory when using qtutils with PySide. \r\n\r\nUnfortunately, this is not a defect in qtutils, but rather a result of a PySide bug:\r\n```\r\n#!python\r\n\r\nQCoreApplication.postEvent()\r\n```\r\nThis bug exists in PySide versions up to 1.2.1 (at least)\r\n\r\nSee [this bug](https://bugreports.qt-project.org/browse/PYSIDE-205) for more details.\r\n\r\nIt will be \"fixed\" when PySide fixed the bug", "markup": "markdown", "html": "<p>Calls to functions in invoke_in_main cause the Python program to leak memory when using qtutils with PySide. </p>\n<p>Unfortunately, this is not a defect in qtutils, but rather a result of a PySide bug:</p>\n<div class=\"codehilite language-python\"><pre><span></span><span class=\"n\">QCoreApplication</span><span class=\"o\">.</span><span class=\"n\">postEvent</span><span class=\"p\">()</span>\n</pre></div>\n\n\n<p>This bug exists in PySide versions up to 1.2.1 (at least)</p>\n<p>See <a data-is-external-link=\"true\" href=\"https://bugreports.qt-project.org/browse/PYSIDE-205\" rel=\"nofollow\">this bug</a> for more details.</p>\n<p>It will be \"fixed\" when PySide fixed the bug</p>", "type": "rendered"}, "assignee": null, "state": "new", "version": null, "edited_on": null, "created_on": "2014-01-23T03:13:14.143337+00:00", "milestone": null, "updated_on": "2018-01-23T03:26:39.950071+00:00", "type": "issue", "id": 1}